iothub/device/iot-device-client/src/main/java/com/microsoft/azure/sdk/iot/device/transport/mqtt/MqttIotHubConnection.java

```diff
@@ -242,6 +242,7 @@ else if (proxySettings.getProxy().type() == Proxy.Type.HTTP)
             this,
             moduleId,
             this.config.getGatewayHostname() != null && !this.config.getGatewayHostname().isEmpty(),
+            this.config.isConnectingToMqttGateway(),
```

Comment:
pass the flag `isConnectingToMqttGateway` to ctor of "MqttMessaging", so the topics can be decided accordingly.